bundles/statistics/statsgrid2016/view/SearchFlyout.js

```diff
@@ -66,28 +66,34 @@ Oskari.clazz.define('Oskari.statistics.statsgrid.view.SearchFlyout', function (t
     getNewSearchElement: function () {
         var me = this;
         var container = jQuery('<div></div>');
-
         var selectionComponent = Oskari.clazz.create('Oskari.statistics.statsgrid.IndicatorSelection', me.instance, me.sandbox);
         container.append(selectionComponent.getPanelContent());
 
+        var buttonContainer = jQuery('<div id="statistics-search-flyout-button-container"></div>');
+        container.append(buttonContainer);
+
         var btn = Oskari.clazz.create('Oskari.userinterface.component.Button');
         btn.addClass('margintopLarge');
         btn.setPrimary(true);
         btn.setTitle(this.loc('panels.newSearch.addButtonTitle'));
         btn.setEnabled(false);
-        btn.insertTo(container);
+        btn.insertTo(buttonContainer);
 
         btn.setHandler(function (event) {
             event.stopPropagation();
-            me.setSpinner(selectionComponent.spinner);
+            const progressSpinner = Oskari.clazz.create('Oskari.userinterface.component.ProgressSpinner');
+            progressSpinner.insertTo(jQuery('#statistics-search-flyout-button-container'));
+            progressSpinner.opts.top = -100;
+            me.setSpinner(progressSpinner);
+            me.getSpinner().start();
```

Comment:
Nothing stops the spinner?

Reply:
Spinner is stopped at  https://github.com/mikkomustikkamaa/oskari-frontend/blob/b58dcaec61779b58dadfa54b49aa0a73248cf3b9/bundles/statistics/statsgrid2016/view/SearchFlyout.js#L522. Mentioned code in _addIndicators function was already implemented so i decided to use it.